l5kit/l5kit/evaluation/metrics.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+
+
+def neg_multi_log_likelihood(
+    gt: np.ndarray, pred: np.ndarray, confidences: np.ndarray, avails: np.ndarray
+) -> np.ndarray:
+    """
+    Compute a negative log-likelihood for the multi-mode scenario.
+    log-sum-exp trick is used here to avoid overflow and overflow, For more information about it see:
+    https://en.wikipedia.org/wiki/LogSumExp#log-sum-exp_trick_for_log-domain_calculations
+    https://timvieira.github.io/blog/post/2014/02/11/exp-normalize-trick/
+    https://leimao.github.io/blog/LogSumExp/
+
+    Args:
+        gt (np.ndarray): array of shape (time)x(2D coords)
+        pred (np.ndarray): array of shape (modes)x(time)x(2D coords)
+        confidences (np.ndarray): array of shape (modes) with a confidence for each mode in each sample
+        avails (np.ndarray): array of shape (time) with the availability for each gt timestep
+    Returns:
+        np.ndarray: negative log-likelihood for this example, a single float number
+    """
+    assert len(pred.shape) == 3, f"expected 3D (MxTxC) array for pred, got {pred.shape}"
+    num_modes, future_len, num_coords = pred.shape
+
+    assert gt.shape == (future_len, num_coords), f"expected 2D (TxC) array for gt, got {gt.shape}"
+    assert confidences.shape == (num_modes,)
+    assert np.allclose(np.sum(confidences), 1), "confidences should sum to 1"
```

Comment:
So I would remove this check, and leave this up to the user. Our metric should by itself be able to distinguish sensible predictions and 'unsensible' ones (e.g. confidences 1, 1, 1).
Maybe the users have a different model in mind which allows these predictions - I would allow it but still 'penalize' predictions we do not expect. 
Also, I see the default tolerance for this is 1e-05 - which I find quite a hard requirement. It might be that a couple of submissions fails because of numerical instabilities.

Reply:
this is a metric related question, let me ask @pondruska about

Reply:
we want to enforce confidences to sum to 1 (I think it's required by the probabilistic framework)

Reply:
1e-5 should be stable enough, but we can simply adjust it if we notice any weird behaviours...

Reply:
Okay.
In that case: How about checks that all elements are in [0, 1] - and e.g. (0, 1, -1) is not valid?
Would that mess up the metric?

Reply:
I asked @pondruska and he said to enforce it to sum to 1